### ISSUE TITLE
b2 upload file is slow

### ISSUE THREAD

**Issue by mdp1** [Backblaze/B2_Command_Line_Tool#405]
Using b2 cli within a backup script.
`b2 upload_file $BUCKET "$path" $DAILY_PATH/$filename`

```
# b2 version
b2 command line tool, version 1.1.0
```

When the backup script is run by cron I seem to consistently get very slow upload speeds (~100kB/s often less).  When the backup script is run directly I often see upload speeds of 1-3MB/s even when run at the same time as the cron is running so slowly. 

![image](https://user-images.githubusercontent.com/12545953/35577413-c8c3b2ae-05e1-11e8-9a4f-40302488a55e.png)

I have seen other b2 cli issues where the upload speed is very slow if number of threads is 1 but gets faster with larger number of threads.  If `--threads` is not specified is there a default or does the b2 cli make a decision automatically which could lead to seeing such a big discrepancy in upload speed for 2 b2 uploads running at the same time?

**Comment by ppolewicz** [comment]
The number of threads defaults to 10 if not specified.

Is the `/root` storage device heavily loaded? If the progress report output cannot be written, it will constraint the performance of the CLI. To check this, you might want to redirect the output from terminal by `./backup.sh > backup-terminal.log`, then tail it and see the performance.

**Comment by mdp1** [comment]
Afaik the /root shouldn't be heavily loaded (server is doing almost nothing at the moment).  However today I can't reproduce: getting 3MB/s upload from the cronned backup script.  I did some network tests yesterday to verify I could push large files to far away servers at at least a few MB/s.  Perhaps b2 was having a bad day yesterday although I am not sure how that explains the significant speed difference between the same script being run at the same time (pushing similar size files to same bucket).  

Writing to files on a slow filesystem is an interesting theory but feels like a bit of a longshot.

Given the problem has magically disappeared today, I would guess that this is most likely an off-server issue either network or with the b2 service.

Googling around, it seems upload speed consistency is a common complaint with b2 (ie people used to S3 notice a big difference using b2).  The daily backups aren't that big for my use case (<1Gbyte) so I guess if it takes a few hours to upload on some days I can live with that for the moment.  

I guess that b2 also only has an endpoint on the west coast so it is possible that there are specific network issues around that endpoint that I couldn't reproduce with the smattering of other servers I did upload tests to.  

Thanks for trying to help.  As the issue is intermittent I will just close this and open a new one if I can reproduce in the future.
